docusaurus/docs/recipes/dropdown.md

```diff
@@ -0,0 +1,427 @@
+
+You can see its documentation [here](https://github.com/downshift-js/downshift#readme) and read [this blog post](https://kentcdodds.com/blog/introducing-downshift-for-react) for further details.
+
+### How it works?
+
+Downshift provides a [`<Downshift>` component](https://github.com/downshift-js/downshift#downshift-component) responsible for providing select and autocomplete/combobox logic as a render prop. It also provides a [set of hooks](https://github.com/downshift-js/downshift#the-react-hooks-api) holding dedicated logic to each behavior.
+
+## Walk-through
+
+### 1. Install `downshift`
+
+```sh
+npm i downshift --save
+```
+
+### 2. Example
+
+Let's say we need to develop a Dropdown component and we have the following data:
+
+```js
+const items = [
+    {
+        label: "January",
+        value: "january"
+    },
+    {
+        label: "February",
+        value: "february"
+    },
+    {
+        label: "March",
+        value: "march"
+    },
+];
+```
+
+We already have our data. What about now? The following code block is a complete example of how to develop a Dropdown component. Check the example first and then follow the explanation to better understand it.
+
+#### 1. Component
+
+ℹ️ We are using [`useSelect` hook](https://github.com/downshift-js/downshift/tree/master/src/hooks/useSelect) to build our example. You can use the [`<Downshift>` component](https://github.com/downshift-js/downshift#downshift-component) as well to achieve the same result. Check an example [here](https://github.com/downshift-js/downshift#usage).
+
+```js
+// Dropdown.js file
+
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { useSelect } from 'downshift';
+import classNames from 'classnames';
+import DropdownItem from 'path-to-dropdown-item';
+
+import styles from './Dropdown.module.css';
+
+const itemToString = (item) => item ? item.value : '';
+const placeholderLabel = 'Choose your option';
+
+const Dropdown = ({ items, onStateChange, className, ...rest }) => {
+    const {
+        isOpen,
+        selectedItem,
+        highlightedIndex,
+        getMenuProps,
+        getItemProps,
+        getToggleButtonProps,
+    } = useSelect({ itemToString, items, onStateChange });
+
+    const renderDropdownItems = useCallback(() => items.map((item, index) => (
+        <DropdownItem
+            item={ item }
+            key={ `dropdown-item-${index}` }
+            selected={ selectedItem === item }
+            highlighted={ highlightedIndex === index }
+            { ...getItemProps({ item, index }) } />
+    )), [highlightedIndex, getItemProps, selectedItem, items]);
+
+    return (
+        <div { ...rest } className={ classNames(styles.container, className) }>
+            <button
```

Comment:
an `id` prop must be provided if we are server rendering downshift. Ref: [comment#429663734](https://github.com/downshift-js/downshift/issues/602#issuecomment-429663734).